app/controllers/paye/PayeErrorController.scala

```diff
@@ -47,9 +57,14 @@ class PayeErrorController @Inject()(
     }
   }
 
-  def authorisedNoAts: Action[AnyContent] = payeAuthAction { implicit request: PayeAuthenticatedRequest[_] =>
+  def authorisedNoAts: Action[AnyContent] = payeAuthAction.async { implicit request: PayeAuthenticatedRequest[_] =>
     {
-      NotFound(payeNoAtsErrorView(PayeAtsMain(payeYear)))
+
+      governmentSpendService.getGovernmentSpendDataV2(payeYear, Some(request.nino)) map { data =>
+        Ok(howTaxIsSpentView(data, payeYear))
+      } recover {
+        case _ => InternalServerError(payeGenericErrorView())
```

Comment:
How come we don't do the same as ATS for IllegalArgumentException here?

`    case e: IllegalArgumentException =>
        logger.error(e.getMessage)
        BadRequest(serviceUnavailableView())`

Also missing logging